common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/dialog/RadioListAlertDialog.kt

```diff
@@ -41,153 +38,138 @@ class RadioListAlertDialog(val builder: Builder) : DialogFragment() {
+            private set
         var selectedOption = 0
+            private set
 
-        fun setTitle(
-            @StringRes textId: Int,
-        ): Builder {
+        fun setTitle(@StringRes textId: Int): RadioListAlertDialogBuilder {
             titleText = context.getText(textId)
             return this
         }
 
-        fun setMessage(
-            @StringRes textId: Int,
-        ): Builder {
+        fun setMessage(@StringRes textId: Int): RadioListAlertDialogBuilder {
             messageText = context.getText(textId)
             return this
         }
 
-        fun setTitle(
-            text: CharSequence,
-        ): Builder {
+        fun setTitle(text: CharSequence): RadioListAlertDialogBuilder {
             titleText = text
             return this
         }
 
-        fun setMessage(
-            text: CharSequence,
-        ): Builder {
+        fun setMessage(text: CharSequence): RadioListAlertDialogBuilder {
             messageText = text
             return this
         }
 
         fun setOptions(
             @StringRes stackedButtonTextId: List<Int>,
             selectedItem: Int = 0
-        ): Builder {
+        ): RadioListAlertDialogBuilder {
             stackedButtonTextId.forEach {
                 optionList.add(context.getText(it))
             }
             selectedOption = selectedItem
             return this
         }
 
-        fun setPositiveButton(@StringRes textId: Int): Builder {
+        fun setPositiveButton(@StringRes textId: Int): RadioListAlertDialogBuilder {
             positiveButtonText = context.getText(textId)
             return this
         }
 
-        fun setNegativeButton(
-            @StringRes textId: Int
-        ): Builder {
+        fun setNegativeButton(@StringRes textId: Int): RadioListAlertDialogBuilder {
             negativeButtonText = context.getText(textId)
             return this
         }
 
-        fun addEventListener(eventListener: EventListener): Builder {
+        fun addEventListener(eventListener: EventListener): RadioListAlertDialogBuilder {
             listener = eventListener
             return this
         }
 
-        fun build(): RadioListAlertDialog {
-            val builder = this
-            if (builder.positiveButtonText.isEmpty()) {
-                throw Exception("TextAlertDialog: You must always provide a Positive Button")
+        fun show() {
+            val binding: DialogSingleChoiceAlertBinding = DialogSingleChoiceAlertBinding.inflate(LayoutInflater.from(context))
+
+            val dialogBuilder = MaterialAlertDialogBuilder(context, com.duckduckgo.mobile.android.R.style.Widget_DuckDuckGo_Dialog)
+                .setView(binding.root)
+                .apply {
+                    setCancelable(false)
+                    setOnDismissListener { listener.onDialogDismissed() }
+                }
+            val dialog = dialogBuilder.create()
+
+            setViews(binding, dialog)
+            checkRequiredFieldsSet()
```

Comment:
this should be done at the beginning of the method, so we avoid inflating layouts if the required fields are missing